File: src/com/train/algorithm/twoPointers/ReverseOnlyLetters.cpp
//
// Created by 罗斯 on 25/8/2022.
//
#include <string>
#include <ctype.h>

using namespace std;
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int i = 0;
        int j = s.length() - 1;
        while(i < j){
            if(isalpha(s[i]) && isalpha(s[j])){
                swap(s[i++],s[j--]);
            }else{
                if (!isalpha(s[i])) ++i;
                if (!isalpha(s[j])) --j;
            }
        }
        return s;
    }
};

File: src/com/train/algorithm/twoPointers/implementInPython/ReverseOnlyLetters.py
class Solution(object):
    def reverseOnlyLetters(self, s):
        list(s)
        start, end = 0, len(s) - 1
        while start < end:
            if s[start].isalpha() and s[end].isalpha():
                s[start], s[end] = s[end], s[start]
                start += 1
                end -= 1
            elif s[start].isalpha() and s[end].isalpha() == False:
                end -= 1
            elif s[start].isalpha() == False and s[end].isalpha():
                start += 1
            else:
                start += 1
                end -= 1
        return "".join(s)
